domains/certificates/Query.API/API.UnitTests/MeasurementsSyncer/SlidingWindowServiceTest.cs

```diff
@@ -0,0 +1,337 @@
+using API.MeasurementsSyncer;
+using DataContext.Models;
+using DataContext.ValueObjects;
+using Measurements.V1;
+using Xunit;
+
+namespace API.UnitTests.MeasurementsSyncer;
+
+public class SlidingWindowServiceTest
+{
+    private readonly SlidingWindowService sut = new();
+    private readonly string GSRN = "123456789123456789";
+    private readonly UnixTimestamp now = UnixTimestamp.Now();
+
+    [Fact]
+    public void GivenSynchronizationPoint_WhenCreatingSlidingWindow_NextFetchIntervalStarsAtSynchronizationPoint()
+    {
+        // Metering point synced up until now
+        var synchronizationPoint = now.RoundToLatestHour().Add(TimeSpan.FromHours(-1));
+        var window = sut.CreateSlidingWindow(GSRN, synchronizationPoint);
+
+        // Get fetch interval
+        var fetchIntervalStart = window.GetFetchIntervalStart();
+
+        // Assert interval starts at synchronization point
+        Assert.Equal(synchronizationPoint, fetchIntervalStart);
+    }
+
+    [Fact]
+    public void GivenMissingMeasurement_WhenCreatingSlidingWindow_FetchIntervalIncludesMissingIntervals()
+    {
+        // Metering point synced up until now with a missing measurement interval
+        var synchronizationPoint = now.RoundToLatestHour().Add(TimeSpan.FromHours(-1));
+        var missingMeasurements = new List<MeasurementInterval>
+            { MeasurementInterval.Create(synchronizationPoint.Add(TimeSpan.FromHours(-3)), synchronizationPoint.Add(TimeSpan.FromHours(-2))) };
+        var window = sut.CreateSlidingWindow(GSRN, synchronizationPoint, missingMeasurements);
+
+        // Get fetch interval
+        var fetchIntervalStart = window.GetFetchIntervalStart();
+
+        // Assert interval starts at missing interval
+        Assert.Equal(synchronizationPoint.Add(TimeSpan.FromHours(-3)), fetchIntervalStart);
+    }
+
+    [Fact]
+    public void GivenMultipleMissingIntervals_WhenCreatingSlidingWindow_FetchIntervalIncludesAllMissingIntervals()
+    {
+        // Metering point synced up until now with a missing measurement interval
+        var synchronizationPoint = now.RoundToLatestHour().Add(TimeSpan.FromHours(-1));
+        var missingMeasurements = new List<MeasurementInterval>
+        {
+            MeasurementInterval.Create(now.Add(TimeSpan.FromHours(-3)), now.Add(TimeSpan.FromHours(-2))),
+            MeasurementInterval.Create(now.Add(TimeSpan.FromHours(-6)), now.Add(TimeSpan.FromHours(-5)))
+        };
+        var window = sut.CreateSlidingWindow(GSRN, synchronizationPoint, missingMeasurements);
+
+        // Get fetch interval
+        var fetchIntervalStart = window.GetFetchIntervalStart();
+
+        // Assert interval starts at earliest missing interval
+        Assert.Equal(now.Add(TimeSpan.FromHours(-6)), fetchIntervalStart);
+    }
+
+    [Theory]
+    [InlineData(false, 2)]
+    [InlineData(true, 0)]
+    public void GivenMeasurements_WhenPublishing_NewMeasurementsShouldBePublished(bool quantityMissing, int publishedCount)
```

Comment:
What is quantityMissing parameter used for?

Reply:
Test if quantityMissing is false, then 2 measurements are published. If quantityMissing is true, then no measurements are published.

Reply:
Yes, so the parameter is used in the test, when creating dummy measurements.

Reply:
In the test example, how can there be a quantity missing in a 1 hour window? I know it's just a test but if it was realistic. Maybe we can talk on teams about it